src/Http/Controllers/FilepondController.php

```diff
@@ -169,14 +174,13 @@ private function persistFileIfDone($disk, $basePath, $length, $finalFilePath)
             // Get chunk contents
             $chunkContents = $storage
                 ->get($chunk);
-
             // Laravel's local disk implementation is quite inefficient for appending data to existing files
             // To be at least a bit more efficient, we build the final content ourselves, but the most efficient
             // Way to do this would be to append using the driver's capabilities
             $data .= $chunkContents;
             unset($chunkContents);
         }
-        Storage::disk($disk)->put($finalFilePath, $data, ['mimetype' => 'application/octet-stream']);
+        Storage::disk($disk)->put($finalFilePath, $data);
```

Comment:
Just removing this `['mimetype' => 'application/octet-stream']` fixed the issue, i know we are saving stream of data but it caused this issue.

Reply:
The reason the mimetype is explicitly set here, is that this avoids a very expensive mime type autodetect operation. Do we have a way to get the proper mime type from filepond? Another solution would be to not build the final file content in memory, but use a temporary file on the disk to assemble the file and upload it from there. Then the mime type check would be more efficient, because it would use php's finfo functionality with that path, instead of working on the whole string in memory.

Reply:
I think we can get the file type from uploaded file using headers function (since we have the file instance and we have its `type` property), but in this case we have to handle the case if we forgot to send it with headers, or if you want we can validate it if it's missing from header we can return an error instead.